tests/test_fitsRawFormatter.py

```diff
@@ -0,0 +1,173 @@
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+
+from astropy.coordinates import Angle
+import astropy.units as u
+
+import lsst.utils.tests
+
+from astro_metadata_translator import FitsTranslator, StubTranslator
+from astro_metadata_translator.translators.helpers import tracking_from_degree_headers
+from lsst.afw.cameraGeom.testUtils import CameraWrapper, DetectorWrapper
+import lsst.afw.geom
+import lsst.daf.base
+import lsst.daf.butler
+import lsst.geom
+from lsst.obs.base import FitsRawFormatterBase, MakeRawVisitInfoViaObsInfo
+from lsst.obs.base.utils import createInitialSkyWcs, InitialSkyWcsError
+
+
+class SimpleTestingTranslator(FitsTranslator, StubTranslator):
+    _const_map = {"boresight_rotation_angle": Angle(90*u.deg),
+                  "boresight_rotation_coord": "sky",
+                  "detector_exposure_id": 12345
+                  }
+    _trivial_map = {"boresight_airmass": "AIRMASS"}
+
+    def to_tracking_radec(self):
+        radecsys = ("RADESYS", )
+        radecpairs = (("RA", "DEC"),)
+        return tracking_from_degree_headers(self, radecsys, radecpairs, unit=(u.deg, u.deg))
+
+
+class MakeTestingRawVisitInfo(MakeRawVisitInfoViaObsInfo):
+    metadataTranslator = SimpleTestingTranslator
+
+
+class SimpleFitsRawFormatter(FitsRawFormatterBase):
+    @property
+    def translatorClass(self):
+        return SimpleTestingTranslator
+
+    def getDetector(self, id):
+        """Use CameraWrapper to create a fake detector that can map from
+        PIXELS to FIELD_ANGLE.
+
+        Always return Detector #10, so all the tests are self-consistent.
+        """
+        return CameraWrapper().camera.get(10)
+
+
+class FitsRawFormatterTestCase(lsst.utils.tests.TestCase):
+    def setUp(self):
+        # The FITS WCS and VisitInfo coordinates in this header are intentionally
+        # different, to make comparisons between them more obvious.
+        self.boresight = lsst.geom.SpherePoint(10., 20., lsst.geom.degrees)
+        self.header = {
+            "AIRMASS": 1.2,
+            "RADESYS": "ICRS",
+            "EQUINOX": 2000,
+            "OBSGEO-X": "-5464588.84421314",
+            "OBSGEO-Y": "-2493000.19137644",
+            "OBSGEO-Z": "2150653.35350771",
+            "RA": self.boresight.getLatitude().asDegrees(),
+            "DEC": self.boresight.getLongitude().asDegrees(),
+            "CTYPE1": "RA---SIN",
+            "CTYPE2": "DEC--SIN",
+            "CRPIX1": 5,
+            "CRPIX2": 6,
+            "CRVAL1": self.boresight.getLatitude().asDegrees() + 1,
+            "CRVAL2": self.boresight.getLongitude().asDegrees() + 1,
+            "CD1_1": 1e-5,
+            "CD1_2": 0,
+            "CD2_2": 1e-5,
+            "CD2_1": 0
+        }
+        # make a property list of the above, for use by the formatter.
+        self.metadata = lsst.daf.base.PropertyList()
+        for key, value in self.header.items():
+            self.metadata.set(key, value)
```

Comment:
```
self.metadata[key] = value
```
should work fine but it might be better to replace the two lines with:
```
self.metadata.update(self.header)
```
which does it all in one go.